C/services/common/plugin_manager.cpp

```diff
@@ -225,3 +226,39 @@ PLUGIN_HANDLE PluginManager::resolveSymbol(PLUGIN_HANDLE handle, const string& s
   return pluginHandleMap.find(handle)->second->ResolveSymbol(symbol.c_str());
 }
 
+/**
+ * Scan "plugins" directory under FOGLAMP_ROOT for the given plugin type:
+ * south, north, filter, notificationRule, notificationDelivery etc
+ *
+ * @param    type		The plugin type
+ * @param    plugins		The output plugin list name to fill	
+ */
+void PluginManager::scanPluginDir(const string& type,
+				  list<string>& plugins)
+{
+	char *home = getenv("FOGLAMP_ROOT");
+	if (home)
+	{
+		struct dirent *entry;
+		DIR *dp;
+		string path = home;
+		path += "/plugins/" + type + "/";
+
+		// Open the plugins dir/type
+		dp = opendir(path.c_str());
+
+		// Get all plugin name directories
+		while ((entry = readdir(dp)))
+		{
+			if (strcmp (entry->d_name, "..") != 0 &&
+			    strcmp (entry->d_name, ".") != 0)
+			{
+				// Load plugin, given its name
+				loadPlugin(entry->d_name, type);
+				// Add name to ouput list
+				plugins.push_back(entry->d_name);
```

Comment:
This is adding the file name, notches plugin name.

Reply:
entry->d_name is the subdirectory name in plugins/type

plugins/filter/
                      delta/libdelta.so
                      scale/libscale.so

taking "delta" and passinf it to loadPlugin() works, at least with C plugins

Another example:

plugins/north/
                  PI_Server_V2/libPI_Server_V2.so

Reply:
ok, thanks